format/FlightSql.proto

```diff
@@ -1450,8 +1484,48 @@ message ActionClosePreparedStatementRequest {
   bytes prepared_statement_handle = 1;
 }
 
+/*
+ * Request message for the "BeginTransaction" action.
+ * Begins a transaction or creates a savepoint within a transaction.
+ */
+message ActionBeginTransactionRequest {
+  // Create a savepoint within the identified transaction. Only supported if
+  // FLIGHT_SQL_TRANSACTION is FLIGHT_SQL_TRANSACTION_SUPPORT_SAVEPOINT.
+  bytes transaction_id = 1;
+  // Name for the savepoint (if applicable).
+  string name = 2;
+}
+
+/*
+ * The result of a "BeginTransaction" action.
+ *
+ * The transaction/savepoint can be manipulated with the "EndTransaction"
+ * action, or automatically via server timeout.
```

Comment:
should server timeout be something that the client has the option of specifying?  Can it be introspected?

Reply:
I chose to follow prepared statements in this regard. I'll add a SqlInfo value to retrieve the timeout.